### PR TITLE
Merge sort

### PR DESCRIPTION
**Pull request by cnickels21**

### CODE REVIEW

**Comment by cnickels21** [comment]
- [x] Top-level README “Table of Contents” is updated
 - [x] Feature tasks for this challenge are completed
 - [x] Unit tests written and passing
     - [ ] “Happy Path” - Expected outcome
     - [x] Expected failure
     - [x] Edge Case (if applicable/obvious)
 - [x] README for this challenge is complete
     - [x] Summary, Description, Approach & Efficiency, Solution
     - [x] Link to code
     - [x] Picture of whiteboard

File: Challenges/Sorts/MergeSort.cs [hunk 1]
```diff
@@ -7,8 +7,109 @@ public class MergeSort
     {
         public static int[] MergeSorter(int[] array)
         {
+            int arrayLength = array.Length;
+
+            if (arrayLength > 1)
+            {
+                int mid = arrayLength / 2;
+                int[] leftSide = new int[mid];
+                int[] rightSide = new int[mid];
```
Comment (TheAgileMCB) [inline Challenges/Sorts/MergeSort.cs:26]: int[] rightSide = new int[arrayLength - mid];

File: Challenges/Sorts/MergeSort.cs [hunk 2]
```diff
@@ -7,8 +7,109 @@ public class MergeSort
...
+
+                for (int i = 0; i < mid; i++)
+                {
+                    leftSide[i] = array[i];
+                }
+
+                for (int i = 0; i < mid; i++)
```
Comment (TheAgileMCB) [inline Challenges/Sorts/MergeSort.cs:33]: for (int i = mid; i < arrayLength; i++)

File: Challenges/Sorts/MergeSort.cs [hunk 3]
```diff
@@ -7,8 +7,109 @@ public class MergeSort
...
+                {
+                    rightSide[i] = array[mid + i];
```
Comment (TheAgileMCB) [inline Challenges/Sorts/MergeSort.cs:35]: rightSide[i - mid] = array[i];

Comment (TheAgileMCB) [reply Challenges/Sorts/MergeSort.cs:35]: I saw this-- that you could introduce a new variable instead of [i-mid] like:

rightSide[x] = array[i];
x++;

File: Challenges/Sorts/MergeSort.cs (hunk 1 above)
Comment (TheAgileMCB) [reply Challenges/Sorts/MergeSort.cs:26]: or add an "if":

if (arrayLength % 2 == 0)
{
right = new int[mid];
}
else
{
rightSide = new int[mid +1]
};

**Review by TheAgileMCB — COMMENTED** [review]
👍
